subprojects/junit-jupiter/src/main/java/org/mockito/junit/jupiter/MockitoExtension.java

```diff
@@ -190,16 +196,24 @@ public void beforeEach(final ExtensionContext context) {
     private void collectParentTestInstances(ExtensionContext context, Set<Object> testInstances) {
         Optional<ExtensionContext> parent = context.getParent();
 
+        boolean firstParent = true;
```

Comment:
Instead of having a boolean here, can we pass in `context.getParent()` on line 164. E.g. we pass the first parent in already, but then we keep all the other logic the same. That seems a little bit nicer to me and easier to follow. The comment can then be placed before line 164 instead.

Reply:
Please push a commit to this branch - it's simpler!

Reply:
Sadly I am not at my working computer. I can do a push on Monday if you prefer.

Reply:
This works :-) Push on Monday. Have a nice weekend!

Reply:
Done PTAL